### DISCUSSION TITLE
Implementing magnetic fields as used defined fuctions

### DISCUSSION THREAD

**Discussion by malpan** [nrc-cnrc/EGSnrc#799] [Questions] [answered]
I'm looking to do some simulations of larger scale MR-Linacs, are there any plans to implement electromagnetic field transport with more than just constant magnetic fields.

I'm willing to build this functionality however I'd appreciate any leads in terms of where to start in the code (EEMF_macros.mortran seems to be the place) and suggestions for algorithms.

**Comment by victorMalkov** [comment, marked as answer]
Hi, the EEMF code would be a good place to start with this. All work conducted with this code has been with constant magnetic fields as this reflected the clinical situations that we were exploring. The code can handle changing magnetic fields since the magnetic field is sampled at each electron step. The limitation in this case would be the distance that the electrons travel compared to the B field gradients - this would mean introducing a step size restriction which depends on the gradient or introducing sampling of the B field along the path of the electron for larger steps. Happy to dive into a deeper discussion off github as well.

**Reply by malpan** [reply]
I have spent the month going through various books on em plasma simulation and have grokked the EEMF macros to a fair degree. 
There's a few interesting modified Boris algorithms around that seem to have good performance in spatially dependent fields. 

I would appreciate your contact details while I work on this, thanks a lot!
